test/commands/config/set.test.js

```diff
@@ -0,0 +1,270 @@
+						newConfig,
+					);
+				});
+
+			setupTest()
+				.stdout()
+				.command([
+					'config:set',
+					'api.nodes',
+					'http://somehost:1234,http://localhost:4000',
+				])
+				.it('should set api.nodes to array with 2 values', () => {
+					const newConfig = {
+						...defaultConfig,
+						api: {
+							network: defaultConfig.api.network,
+							nodes: ['http://somehost:1234', 'http://localhost:4000'],
+						},
+					};
+					return expect(config.setConfig).to.be.calledWith(
+						defaultDir,
+						newConfig,
+					);
+				});
+
+			setupTest()
+				.stdout()
+				.command(['config:set', 'api.nodes'])
+				.it('should set api.nodes to empty array', () => {
+					const newConfig = {
+						...defaultConfig,
+						api: {
+							network: defaultConfig.api.network,
+							nodes: [],
+						},
+					};
+					return expect(config.setConfig).to.be.calledWith(
+						defaultDir,
+						newConfig,
+					);
+				});
+
+			setupTest()
+				.stdout()
+				.command(['config:set', 'api.nodes', 'ws://hostname'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'Node URLs must include a supported protocol',
+					),
+				)
+				.it(
+					'should throw error when api.nodes value is not supported protocol',
+				);
+
+			setupTest()
+				.stdout()
+				.command(['config:set', 'api.nodes', 'http://'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'Node URLs must include a supported protocol (http, https) and a hostname.',
+					),
+				)
+				.it('should throw error when api.nodes value does not have host name');
+
+			setupTest()
+				.stdout()
+				.stub(config, 'getConfig', sandbox.stub().returns({}))
+				.command(['config:set', 'api.nodes', 'http://hostname'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'It looks like your configuration file is corrupted. Please check the file at',
+					),
+				)
+				.it('should throw error when config file is corrupted');
+		});
+
+		describe('config:set json value', () => {
+			setupTest()
+				.stdout()
+				.command(['config:set', 'json', 'true'])
+				.it('should set json to provided boolean value', () => {
+					const newConfig = {
+						...defaultConfig,
+						json: true,
+					};
+					return expect(config.setConfig).to.be.calledWith(
+						defaultDir,
+						newConfig,
+					);
+				});
+
+			setupTest()
+				.stub(config, 'setConfig', sandbox.stub().returns(false))
+				.stdout()
+				.command(['config:set', 'json', 'true'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'Config file could not be written: your changes will not be persisted',
+					),
+				)
+				.it('should throw an error when setConfig fails');
+
+			setupTest()
+				.stdout()
+				.command(['config:set', 'json', 'truely'])
+				.catch(error =>
+					expect(error.message).to.contain('Value must be a boolean.'),
+				)
+				.it('should throw error when json value is not boolean');
+		});
+
+		describe('config:set name value', () => {
```

Comment:
Do we still need this with oclif?

Reply:
it's a feature u can name ur commander, which does nothing =)
let's delete this 😢